walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,220 @@
+        if (mandatoryAttendees.contains(attendee)) {
+          TimeRange currEventTimeRange = event.getWhen(); 
+        
+          handleOverlapTimeRange(currEventTimeRange, busyTimes);
+
+          // If any one of the mandatory attendees is in this event's guest list,
+          // this event's time range needs to be considered busy. 
+          // Breaks out of the for-each loop because there is no need to look
+          // at the rest of the event's guest list. 
+          break; 
+        }
+      }
+    }); 
+    return busyTimes; 
+  }
+
+  /**
+   * Compares a given event's time slot with all the existing busy time slots.
+   * Merges the current event time slot with a busy time slot if they overlap.
+   * Adds the current event's time slot to the list of busy slots in the end. 
+   * @param current event's time range; a collection of busy time slots
+   */ 
+  private void handleOverlapTimeRange(TimeRange currEventTimeRange, Collection<TimeRange> busyTimes) {
+    boolean overlapped = false; 
+
+    for (TimeRange busyTimeRange : busyTimes) {
+      if (currEventTimeRange.overlaps(busyTimeRange)) {
+        TimeRange combinedTimeRange = getCombinedTimeRange(currEventTimeRange, busyTimeRange);
+        busyTimes.remove(busyTimeRange);
+        busyTimes.add(combinedTimeRange);
+        overlapped = true;  
+        currEventTimeRange = combinedTimeRange; 
+      } 
+    }
+
+    // If the current event doesn't overlap with any existing busy time range,
+    // add the current event's time range to the set of all busy time slots. 
+    if (!overlapped) {
+      busyTimes.add(currEventTimeRange); 
+    }
+  }
+
+  /** Creates a new overall combined time range for two time ranges that overlap. */ 
+  private TimeRange getCombinedTimeRange(TimeRange a, TimeRange b) {
+    int aStart = a.start();
+    int aEnd = a.end();
+    int bStart = b.start();
+    int bEnd = b.end();
+
+    TimeRange combinedTimeRange = TimeRange.fromStartEnd(
+        /* start time = */ Math.min(aStart, bStart), 
+        /* end time = */ Math.max(aEnd, bEnd),
+        /* inclusive of end time = */ false); 
+
+    return combinedTimeRange; 
+  }
+
+  /** Sorts a list of busy time ranges by their start time. */
+  private void sortBusyTimeRangesByStart(List<TimeRange> allBusyTimeRanges) {
+    Collections.sort(allBusyTimeRanges, TimeRange.ORDER_BY_START); 
+  }
+
+  /**
+   * Finds all empty time ranges based on a list of busy time ranges and 
+   * the length of the requested meeting
+   */ 
+  private List<TimeRange> getEmptyTimeRanges(List<TimeRange> allBusyTimeRanges, long meetingDuration) {
+    List<TimeRange> emptyTimeRanges = new ArrayList<TimeRange>();
+
+    if (allBusyTimeRanges.size() == 0 && meetingDuration <= END_OF_DAY_MINUTES) {
+      emptyTimeRanges.add(TimeRange.fromStartEnd(
+          /* start time = */ 0, 
+          /* end time = */ END_OF_DAY_MINUTES, 
+          /* inclusive of end time = */ false));
```

Comment:
I noticed that the TimeRange class already has START_OF_DAY and END_OF_DAY defined as constants, and in that class, end of day is defined as 1439, which is the number of minutes for 23:59. I changed my code so that these two constants from TimeRange are used instead of my own defined constants for 0 and 1440. Based on TimeRange class's definition of the end of the day, I changed the "inclusive" parameter to true whenever the end time is TimeRaneg.END_OF_DAY. I also added a test to confirm my code change works. In my test, a 24-hour meeting is defined as having 1439 minutes.